### PR TITLE
fix(release): allow downloading binary via proxy again

### LINKED ISSUE #402: Builds v5.0.0-5.1.1 include lots of suspicious looking, unnecessary code
Looking at upgrading a project that is using 4.0.2, and I noticed the 5.x releases were out. Trying to find the change log, I noticed that npmjs.com lists a steady increase in file size. Got suspicious, and it looks like the dist includes a ton of stuff that doesn't belong there!

4.0.2 - 19.5kB: 2 small js files plus standard package.json/README/LICENSE/CHANGELOG files
5.0.0 - 782kB: a dist folder with a single 776kB index.js, looks like it has graphql clients, fetch wrappers, lots of other stuff included
5.0.1 - 782kB: similar to 5.0.0
5.1.0 - 3.27MB: dist folder includes 3.17MB index.js, setup-node-sandbox.js, setup-sandbox.js, events.js, bridge.js, 37.index.js, all sorts of strange code in them.
5.1.1 - 3.27MB: similar to 5.1.0

Even if the extra code wasn't suspicious and the actual utility ran fine, I am not willing to introduce 3MB into my repos!

Please investigate what is causing this, clean up the build, and release a version which doesn't look like a security nightmare.

### LINKED ISSUE #405: 5.1.2 no longer supports HTTP proxy
Just to confirm https://github.com/editorconfig-checker/editorconfig-checker.javascript/issues/402#issuecomment-1816391363, 5.1.2 no longer downloads the go binary via proxy:

```
$ yarn lint:editorconfig
yarn run v1.22.19
$ editorconfig-checker docs/*/*.md docs/*.md *.md
Failed to download binary:
HttpError: fetch failed
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

(follow-up to https://github.com/editorconfig-checker/editorconfig-checker.javascript/issues/394)

We re-added this earlier this year in https://github.com/editorconfig-checker/editorconfig-checker.javascript/pull/395, so might just need some tweaking.

/cc @theoludwig

### PR DESCRIPTION
**Pull request by nejch**
# What changes this PR introduce?

I managed to find a way to test this, plus using the new undici methods and agents. This coincidentally solves #402, since the built javascript is back to 1MB.

The new undici/octokit has absolutely no support for standard environment variables for proxies, so I had to add a rather old `proxy-from-env` library, but in the end I realized that's what the old proxy-agent library uses anyway - and in fact it's a solved problem so I don't see it being from 2020 a real issue.

I reused octokit's approach to proxy testing with a real little proxy server, this should help if people refactor code to still tes something close to a real scenario.

Edit: since I added some real tests, this refactors the naming of the script to be a bit more accurate.

I also had to bump tsx due to CI failures (see e.g. https://github.com/redwoodjs/redwood/pull/9653)

## List any relevant issue numbers

Closes #405 
Closes #402 (node-fetch and proxy-agent replaced with undici's implementations)

## Is there anything you'd like reviewers to focus on?

Not sure if we should add a note to not delete those tests to avoid any further regressions :sweat_smile:

### CODE REVIEW

File: src/release.ts [hunk 1]
```diff
@@ -51,3 +50,19 @@ function getAssetPrefix() {
   }
   return `ec-${platform}-${arch}`
 }
+
+// eslint-disable-next-line  @typescript-eslint/no-explicit-any
```
Comment (nejch) [inline src/release.ts:54]: I might have been a bit lazy here, feel free to correct me with a better idea :sweat_smile:

**Comment by nejch** [comment]
@theoludwig @mstruebing let me know what you think! I tested this already in our corporate enviornment and should work fine.

**Review by theoludwig — DISMISSED** [review]
Thank you for the PR. :rocket:

Mostly looks good to me.
I will try to test it locally, this afternoon.
We might find a way to refactor to avoid the `any` and maybe we can directly use the built-in Node.js fetch (which uses `undici` under the hood), but that's not a blocker to merge and release this if we can't do the refactor. :smile:

**Comment by nejch** [comment]
Thanks for the quick review @theoludwig! 

Ah ok, I'm just not sure if the bundled undici exposes ProxyAgent, so I just imported fetch explicitly along with it. I still need to dive a bit more into the ecosystem though :)

Also apologies if I'm mixing styles a bit here. just noticed and pushed, maybe needs more linting for people like me :sweat_smile:

**Comment by theoludwig** [comment]
@nejch I refactored the code, to avoid the `any` type, by importing `RequestInit` type from `undici`.

About using Node.js built-in fetch for `ProxyAgent`, it is not yet possible. So we must still use `undici`, but apparently in future Node.js versions, we will maybe be able to, there is an open issue here: https://github.com/nodejs/node/issues/43187
When `undici` will directly be exposed with Node.js, we should be able to use it, and again reduce again the package size.
But for now, it is good enough, 1MB. :smile:

> Also apologies if I'm mixing styles a bit here. just noticed and pushed, maybe needs more linting for people like me 😅

 No worries, I formatted the code with Prettier, and now, it is also verified with CI.

**Review by theoludwig — APPROVED** [review]

**Pull request merged**